fenicsxconcrete/experimental_setup/base_experiment.py

```diff
@@ -22,36 +23,43 @@ def __init__(self, parameters: dict[str, pint.Quantity]):
         """
 
         # initialize parameter attributes
-        self.parameters = parameters
+        default_setup_parameters = Parameters()
+        # setting up default setup parameters
+        default_setup_parameters['degree'] = 2 * ureg('')  # polynomial degree
+
+        # update with input parameters
+        default_setup_parameters.update(parameters)
+        # as attribute
+        self.parameters = default_setup_parameters
+        # remove units for use in fem model
         self.p = self.parameters.to_magnitude()
 
         self.setup()
 
     @abstractmethod
     def setup(self):
         """Is called by init, must be defined by child"""
-        pass
 
     @staticmethod
     @abstractmethod
     def default_parameters() -> dict[str, pint.Quantity]:
         """returns a dictionary with required parameters and a set of working values as example"""
-        # this must de defined in each setup class
-        pass
 
+    @abstractmethod
     def create_displacement_boundary(self, V) -> list:
-        # define empty displacement boundary
-        displ_bcs = []
+        """returns a list with displacement boundary conditions
 
-        return displ_bcs
+           this function is abstract until there is a need for a material that does not need a displacement boundary
+           once that is required, just make this a normal function that returns an empty list
+           """
 
-    def create_force_boundary(self, v):
+    def create_force_boundary(self, v = None):
         # define empty force boundary
         # TODO: is there a better solution???
 
         return None
```

Comment:
hmm, the `BoundaryConditions` class is supposed to be able to handle this.
Do we want to use it here?

Reply:
@pdiercks: I would suggest to use #29 for this, or should we focus on the displacement boundaries here and open a new issue for the force boundary implementation?

Reply:
yes, #29 is probably the better place to deal with that.